website/content/userguide/permanent-data.markdown

```diff
@@ -0,0 +1,120 @@
+---
+title: "Access permanent data"
+menuTitle: "Access permanent data"
+weight: 49
+---
+
+## Create the filesystem
+
+gokrazy provides a handy way to create the permanent data filesystem on the forth partition of your gokrazy installation.
+The [gokrazy/mkfs](https://github.com/gokrazy/mkfs) program will do all the required steps to create the filesystem.
+To actually access the permanent data partition from your own program, gokrazy will mount the partition under `/perm` directory during the startup.
+
+You can either include this program in your `gokr-packer` command line:
+
+```
+gokr-packer \
+  -overwrite=/dev/sdx \
+  -serial_console=disabled \
+  github.com/gokrazy/fbstatus \
+  github.com/gokrazy/hello \
+  github.com/gokrazy/serial-busybox \
+  github.com/gokrazy/mkfs
+```
+
+…or, if you want to run it only once without otherwise including it in your
+installation, you can use `gok run`:
+
+```
+git clone https://github.com/gokrazy/mkfs
+cd mkfs
+gok run -i bakery
+```
+
+```bash
+sudo mkfs.ext4 /dev/sdc
+```
+
+## Permanent data in action
+
+The Go program demonstrates the functionality of the permanent data mount:
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+)
+
+const config = "/perm/config.json"
+
+type Config struct {
+	FilesToCreate int `json:"files_to_create"`
+}
+
+func main() {
+	_, err := os.Stat(config)
+	if err != nil {
+		fsErr := &fs.ErrNotExist
+		if !errors.As(err, fsErr) {
+			panic(err)
+		}
+
+		content, err := json.Marshal(&Config{FilesToCreate: 4})
+		if err != nil {
+			panic(err)
+		}
+
+		if err = ioutil.WriteFile(config, content, 0600); err != nil {
+			panic(err)
+		}
+	}
+
+	content, err := ioutil.ReadFile(config)
+	if err != nil {
+		panic(err)
+	}
+
+	var c Config
+	if err := json.Unmarshal(content, &c); err != nil {
+		panic(err)
+	}
+
+	for i := 0; i < c.FilesToCreate; i++ {
+		if err := ioutil.WriteFile(fmt.Sprintf("/perm/%d.txt", i), []byte("gokrazy rocks"), 0600); err != nil {
+			panic(err)
+		}
+	}
+}
+
+```
+
+## Manual inspection/modification
+
+You can inspect and modify the data on the permanent data filesystem manually.
+This can be handy if you need e.g. to pass a configuration file for your application.
+To access the filesystem put the SD card into your PC/notebook. Your OS will likely mount the filesystem automatically.
+
+gokrazy will access your files as `root`, which means you will have to set proper permissions before placing data on the filesystem.
```

Comment:
Why? You shouldn’t need to have to care about ownership, as root can always access files regardless of their ownership or permission:
```
/tmp/breakglass422977765 # stat /perm/demo.txt
  File: /perm/demo.txt
  Size: 4         	Blocks: 8          IO Block: 4096   regular file
Device: b304h/45828d	Inode: 27          Links: 1
Access: (0600/-rw-------)  Uid: (65534/ UNKNOWN)   Gid: (65534/ UNKNOWN)
Access: 2022-01-11 21:32:11.000000000
Modify: 2022-01-11 21:32:11.000000000
Change: 2022-01-11 21:32:22.000000000

/tmp/breakglass422977765 # cat /perm/demo.txt
wow

```

I think you can simplify this entire section of the document to just copy files over to the permanent data partition without caring about file ownership.